### PR TITLE
Implemented `RobertaForSequenceClassification`

### PR DESCRIPTION
**Pull request by Seun-Ajayi**
## Proposed changes

Implemented `RobertaForSequenceClassification` in the `roberta.py` module. Added `RobertaClassificationHead` along the line and added `SequenceClassifierOutput` class to the `modelling_outputs.py` script since it is the output for the sequence classifier. 

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [x] New Model Architecture
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

For `multi_label_classification` problem type, `BCEWithLogitsLoss` is used but this is not present in mlx-core module  
 ```
            elif self.config.problem_type == "multi_label_classification":
                loss_fct = BCEWithLogitsLoss()
                loss = loss_fct(logits, labels)

```
so I had to use `binary_cross_entropy`
```
            elif self.config.problem_type == "multi_label_classification":
                loss_fct = nn.losses.binary_cross_entropy()
                loss = loss_fct(logits, labels)
```

How else could I have handled this?

### CODE REVIEW

**Comment by ToluClassics** [comment]
Can you please add a test case?

**Comment by Seun-Ajayi** [comment]
Implemented `RobertaForTokenClassification` and `RobertaForQuestionAnswering` too.

`CrossEntropyLoss` method takes `ignore_index` as an argument but this parameter is not included in mlx's version of it, so I just did without the parameter. This is in  the `RobertaForQuestionAnswering` class
```
            loss_fct = CrossEntropyLoss(ignore_index=ignored_index)
```
My workaround
```
            loss_fct = nn.losses.cross_entropy()

```

File: src/mlx_transformers/models/roberta.py [hunk 1]
```diff
@@ -451,6 +451,252 @@ def __call__(
...
+class RobertaForQuestionAnswering(nn.Module):
+    def __init__(self, config):
+        super().__init__(config)
+        self.num_labels = config.num_labels
+
+        self.roberta = RobertaModel(config, add_pooling_layer=False)
+        self.qa_outputs = nn.Linear(config.hidden_size, config.num_labels)
+
+        # Initialize weights and apply final processing
+        self.post_init()
```
Comment (ToluClassics) [inline src/mlx_transformers/models/roberta.py]: needs to go

File: src/mlx_transformers/models/roberta.py [hunk 2]
```diff
@@ -451,6 +451,252 @@ def __call__(
...
+
+        self.roberta = RobertaModel(config, add_pooling_layer=False)
+        classifier_dropout = (
+            config.classifier_dropout if config.classifier_dropout is not None else config.hidden_dropout_prob
+        )
+        self.dropout = nn.Dropout(classifier_dropout)
+        self.classifier = nn.Linear(config.hidden_size, config.num_labels)
+
+        # Initialize weights and apply final processing
+        self.post_init()
```
Comment (ToluClassics) [inline src/mlx_transformers/models/roberta.py]: needs to go

File: src/mlx_transformers/models/roberta.py [hunk 3]
```diff
@@ -451,6 +451,252 @@ def __call__(
...
-        pass
+        super().__init__(config)
+        self.num_labels = config.num_labels
+        self.config = config
+
+        self.roberta = RobertaModel(config, add_pooling_layer=False)
+        self.classifier = RobertaClassificationHead(config)
+
+        # Initialize weights and apply final processing
+        self.post_init()
```
Comment (ToluClassics) [inline src/mlx_transformers/models/roberta.py]: needs to go

**Comment by Seun-Ajayi** [comment]
> Can you please add a test case?

I would

**Comment by Seun-Ajayi** [comment]
We are good now

**Review by ToluClassics — APPROVED** [review]

**Comment by ToluClassics** [comment]
> We are good now

Thank you!

**Pull request merged**